### PR TITLE
LOD: Add hysteresis option.

### LINKED ISSUE #14565: LOD: Consider adding hysteresis option.
With the LOD systems in Unreal, Unity, and Blender, LODs have a threshold that offsets the distance where the LOD appears from the distance where it disappears, to reduce flickering from small movements (e.g. head movement with roomscale VR). From [Unreal docs](https://docs.unrealengine.com/en-us/Engine/Animation/Persona/MeshDetails):

> **LOD Hysteresis** | Used to avoid "flickering" when on LOD boundry. Only take into account when moving from complex to simple.

### PR DESCRIPTION
**Pull request by donmccurdy**
Fixes #14565.

### CODE REVIEW

**Comment by donmccurdy** [comment]
Closing per https://github.com/mrdoob/three.js/issues/14565#issuecomment-410729010, but glad to reopen if others have a use for this.

**Comment by mrdoob** [comment]
Lets keep this open 😊

**Comment by Mugen87** [comment]
Is it possible to rebase and merge this PR? 😇 

I've recently used the exact same code to avoid the LOD-flickering mentioned in #14565.

**Comment by donmccurdy** [comment]
✅ Rebased! 

frankenstein-its-alive

**Review by Mugen87 — APPROVED** [review]

File: test/unit/src/objects/LOD.tests.js [hunk 1]
```diff
@@ -78,9 +78,9 @@ export default QUnit.module( 'Objects', () => {
 			lod.addLevel( low, 50 );
```
Comment (gsimone) [inline test/unit/src/objects/LOD.tests.js:78]: You could add a test for hysteresis in this by adding one similar to the `getObjectForDistance` test but where you stay around the hysteresis range and make sure no change happens until you get out of it

Comment (donmccurdy) [reply test/unit/src/objects/LOD.tests.js:78]: Good point - will add a unit test if this ends up being merged 👍

File: src/objects/LOD.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ class LOD extends Object3D {
 
 	}
 
-	addLevel( object, distance = 0 ) {
+	addLevel( object, distance = 0, hysteresis = 0.05 ) {
```
Comment (Mugen87) [inline src/objects/LOD.js:51]: Revisiting this PR.

What do you think about changing the default of `hysteresis` to `0`? Having a hysteresis option is great but adding it now should probably not affect current setups. Meaning I would not "activate" it by default but let the user define an appropriate value.

Comment (donmccurdy) [reply src/objects/LOD.js]: Agreed and done!

**Pull request merged**